### ISSUE TITLE
Why saving txt_labels with no bbox!!

### ISSUE THREAD

**Issue by majnas** [ultralytics/ultralytics#2067]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I ran prediction on video using following code, the output folder contain all txt_files for all frame (even those with no bbox). I think the txt file only for the frames with bbox would be much more suitable :confused: 

```python
    # Load a model
    model = YOLO(model_path)  # load a custom model
    # Predict with the model
    results = model("/home/dev/Desktop/GH011920.avi", stream=True, save_txt=True, conf=conf, project=project, name=name, save_conf=True) 

```

```bash
(av) dev@av1: /home/dev/Desktop/GH011920  
➜   tree -h | head
[4.0K]  .
└── [260K]  labels
    ├── [   0]  GH011920_1000.txt
    ├── [   0]  GH011920_1001.txt
    ├── [   0]  GH011920_1002.txt
    ├── [   0]  GH011920_1003.txt
    ├── [   0]  GH011920_1004.txt
    ├── [   0]  GH011920_1005.txt
    ├── [   0]  GH011920_1006.txt
    ├── [   0]  GH011920_1007.txt
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@majnas thank you for bringing this to our attention. I understand your concern and agree that it would be more suitable to only have txt files for frames with bounding boxes. 
One way to implement this is to modify the code in YOLOv8 to only save txt files for frames with bounding boxes. This could also potentially improve the performance of the prediction process.
You can create a pull request with your modifications and we will be happy to review and merge it if it meets our standards. Additionally, please note that the YOLOv8 documentation at https://docs.ultralytics.com/ has more information on this topic.

**Comment by Laughing-q** [comment]
@majnas yeah nice catch! thanks! fixed in [https://github.com/ultralytics/ultralytics/pull/2080](https://github.com/ultralytics/ultralytics/pull/2080)

**Comment by majnas** [comment]
> @majnas thank you for bringing this to our attention. I understand your concern and agree that it would be more suitable to only have txt files for frames with bounding boxes. One way to implement this is to modify the code in YOLOv8 to only save txt files for frames with bounding boxes. This could also potentially improve the performance of the prediction process. You can create a pull request with your modifications and we will be happy to review and merge it if it meets our standards. Additionally, please note that the YOLOv8 documentation at https://docs.ultralytics.com/ has more information on this topic.

Thanks, Will do that for next bug/feature! Thanks for your hard work :+1:

**Comment by majnas** [comment]
> @majnas yeah nice catch! thanks! fixed in #2080

No worries, Keep fixing and improving guy, you're brilliant :+1:

**Comment by glenn-jocher** [comment]
Thank you for your kind words, @majnas! Our team is committed to continuous improvement and we are always working on fixing issues and adding new features. If you have any other suggestions or concerns, please don't hesitate to let us know. We greatly appreciate your support and look forward to hearing more from you in the future.
